sympy/integrals/integrals.py

```diff
@@ -436,6 +438,8 @@ def doit(self, **hints):
             # See the integrate() docstring for details.
             def try_meijerg(function, xab):
                 ret = None
+                function = function.xreplace(dict([(h, h.rewrite(exp))
+                    for h in function.atoms(HyperbolicFunction)]))
```

Comment:
I originally tried this in the `_guess_expansion` but some of the tests I added failed if it was done that late. I'll try again to see if there is a better place withing meijer itself.